lib/webhooks/base.rb

```diff
@@ -0,0 +1,35 @@
+module Webhooks
+  class Base
+    attr_reader :request, :repository
+
+    def initialize(request, repository)
+      @request = request
+      @repository = repository
+    end
+
+    def valid_address?
+      repository.address == webhook_address
+    end
+
+    def valid_branch?
+      repository.branch == webhook_branch
+    end
+
+    def valid_request?; raise NotImplementedError; end
+
+    def webhook_address; raise NotImplementedError; end
+
+    def webhook_branch; raise NotImplementedError; end
+
+    protected
+
+    def branch_from_ref(ref)
```

Comment:
Would be nice if you could provide a short description of what the method expects to be a `ref`